js/views/userPageVw.js

```diff
@@ -307,6 +307,25 @@ module.exports = baseVw.extend({
           self.model.set({ownPage: self.options.ownPage});
           self.render();
           !self.currentItemHash && self.loadingDeferred.resolve();
+          
+          // Handle was requested
+          if (self.model.get('page').profile.handle) {
```

Comment:
The router only processes handles if someone enters @<username> in the address bar.

So, the userPage is a more comprehensive place to get the handle, since it will get it whether you're accessing the page via @<username>, guid, etc... (and clicking on a username link usually accesses the user view via guid, so the router wouldn't pick that up).

The only thing I would consider changing is instead of storing the logic to record handles in the userPage, I would have the userPage fire an event when a handle is retrieved and then catch that event in pageNavView and record the handles there. The benefit of that is that you are centralizing all the functionality dealing with storing and retrieving the handles from LS in one place.